docs(howto): add custom domain to functions MTA-1777 (#426)
scaleway/docs-content: compute/functions/how-to/add-a-custom-domain-name-to-a-function.mdx

---
meta:
  title: How to add a custom domain to a function

>   description: This page explains how toadd a custom domain to a function

Reviewer: ```suggestion
  description: This page explains how to add a custom domain to a function
```

content:

>   h1: How toadd a custom domain to a function

Reviewer: ```suggestion
  h1: How to add a custom domain to a function
```

  paragraph: This page explains how to add a custom domain to a function
dates:
  validation: 2022-02-21
  posted: 2022-02-21
---

> Each function comes with an automatically generated domain on which it is available. However it is possible to use your own domain name to address your functions to facilitate their use.

Reviewer: ```suggestion
Each function comes with an automatically generated domain on which it is available. However it is possible to assign your own domain name to make it easier to use your functions.
```

<Message type="requirement">

- You have an account and are logged into the [Scaleway console](https://console.scaleway.com)

> - A running [Serverless Function](https://www.scaleway.com/en/docs/compute/functions/how-to/deploy-a-function/)

Reviewer: ```suggestion
- You have a running [Serverless Function](https://www.scaleway.com/en/docs/compute/functions/how-to/deploy-a-function/)
```

> - A domain name 

Reviewer: ```suggestion
- You have a domain name 
```

</Message>

> 1. Add a CNAME pointing to your function default URL from the [domains dashboard](/network/dns-cloud/how-to/manage-dns-records/).

Reviewer: ```suggestion
1. Add a CNAME record pointing to your function's default URL from the [domains dashboard](/network/dns-cloud/how-to/manage-dns-records/).
```

Reviewer: Does the domain DNS have to be managed by SCW?

Author: I dont't think it has forcefully be managed by scw DNS - we can generalize this if you prefer

> 2. Click the **Endpoints** tab on your Functions' page.

Reviewer: ```suggestion
2. Click the **Endpoints** tab on your Function's Overview page.
```

  <Lightbox src="scaleway-functions-domain.png" alt="" />
3. Click **+ Add a New Endpoint**.

> 4. Enter your custom URL and validate the form.

Reviewer: ```suggestion
4. Enter your custom domain URL and validate the form.
```
